### PR TITLE
integration test steps for the Secured Spring Boot mission

### PR DESCRIPTION
**Pull request by inoxx03**
NOTE: This might need to be rebased after  #750 has been merged.

### CODE REVIEW

File: docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc [hunk 1]
```diff
@@ -1,32 +1,57 @@
...
 * The `oc` client authenticated.
+// Should probably include the prerequisite to create a separate project for IT's in OpenShift
 
 .Procedure
 
+ifdef::secured-spring-boot-mission[]
+
+[IMPORTANT]
+--
+Keep the contents of the `service.sso.yaml` and the `.openshiftio/service.sso.yaml` templates synchronized to ensure that the integration tests work correctly.
```
Comment (Ladicek) [inline docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc]: Do we actually expect end users to modify these files? This is just a note for booster authors I believe.

Comment (inoxx03) [reply docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc]: @Ladicek You're right. I don't think users need to modify them in any way. If you see no reason for us to keep the not, I'm ok with removing it.

File: docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc [hunk 2]
```diff
@@ -1,32 +1,59 @@
 
-[#running-the-secured-booster-integration-tests-on-vertx]
-= Running the {name-mission-secured} Booster Integration Tests on {VertX}
+[#running-the-secured-booster-integration-tests-on-{parameter-runtime}]
+= Running the {name-mission-secured} Booster Integration Tests on {runtime}
+//TODO: specify parameter in master file
```
Comment (tradej) [inline docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc]: Please remove :)

File: docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc [hunk 3]
```diff
@@ -1,32 +1,59 @@
...
 
 .Prerequisites
 
+
 * The `oc` client authenticated.
+// Should probably include the prerequisite to create a separate project for IT's in OpenShift
```
Comment (tradej) [inline docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc]: If the process requires it, I certainly concur.

File: docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc [hunk 4]
```diff
@@ -1,32 +1,59 @@
...
+////
+ifdef::secured-spring-boot-mission[]
+
+[IMPORTANT]
+--
+Keep the contents of the `service.sso.yaml` and the `.openshiftio/service.sso.yaml` templates synchronized to ensure that the integration tests work correctly.
+--
+
+endif::secured-spring-boot-mission[]
+////
```
Comment (tradej) [inline docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc]: Please remove.

File: docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc (hunk 2 above)
Comment (inoxx03) [reply docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc]: Ah, thanks, will do:)

File: docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc (hunk 3 above)
Comment (inoxx03) [reply docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc]: IIRC, it used to be the case that running integration tests inside an OpenShift project already containing a booster application resulted in the removal of this pre-existig application once the test were done. If that still apllies, using a separate project to run the IT's could  help  avoid accidental deletion of a running application. We could either include a step to create an extra project or include a note about it to alert the users to this possibility. @Ladicek WDYT?

File: docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc [hunk 5]
```diff
@@ -1,10 +1,11 @@
 
-[#running-the-secured-booster-integration-tests-on-vertx]
-= Running the {name-mission-secured} Booster Integration Tests on {VertX}
+[#running-the-secured-booster-integration-tests-on-{parameter-runtime}]
```
Comment (tradej) [inline docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc]: Sorry, I forgot, can you please add the following comment to the file?

```
// This is a parameterized module. Parameters used: 
// 
//   parameter-$SOMETHING: $DESCRIPTION_OF_THE_PARAMETER
//   parameter-$SOMETHING_ELSE: ...
//
// Rationale: $YOUR_RATIONALE
```

I will merge it then.

File: docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc (hunk 3 above)
Comment (Ladicek) [reply docs/topics/proc_running-the-secured-booster-integration-tests-on-sb-vx.adoc]: So, if all the boosters migrated their tests to Arquillian Cube, there would be a very simple answer: verify that you have permissions to create new projects in OpenShift and add `-Dnamespace.use.current=false -DenableImageStreamDetection=true` when running the tests. With these parameters, Arquillian Cube would automatically create an ephemeral OpenShift project, run the tests in it and then destroy it.

Unfortunately, some boosters still use OpenShiftTestAssistant for their tests :-/

I think adding a note about the possibility to run the tests in an extra namespace is a good idea.

**Comment by inoxx03** [comment]
@tradej added the comment explaining parameters, please let me know what you think.

**Comment by tradej** [comment]
Cool, thanks. Merging...

**Comment by tradej** [comment]
merged: 8036ac6a2

**Pull request closed without merging**